addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1424,6 +1422,7 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
     _onClick: function (ev) {
         // Do not close the colorpalette on colorpicker click
         if (!ev.originalEvent.__isColorpickerClick) {
+            this._renderColorPalette();
```

Comment:
This change really seems not robust.

1) Here we call the render when the element is clicked but not inside it... feels like you are re-rendering when the colorpicker is being opened *and* closed, is not it? (= why not overriding the `open` method instead of the `_onClick` one?)

2) `_renderColorPalette` is async, calling it without caring anywhere about its async part seems really dangerous, especially keeping future improvements / custo in mind. I feel like this is in general the cause of race condition, flickering and other difficult-to-reproduce issues. It's ok to not wait for all async parts in the rendering flow, but in general it should still be awaited *somewhere*.

3) In `setValue` where you removed the call, the next line does `this.colorPalette.getColorNames()` and this call only makes sense if the `colorPalette` has been fully rendered (see the `colorNames` being initialized in the start method). Probably ok "because it is not really async anymore at that time" but still not robust (future custo, extension, whatever) (= what could be done to be more robust is to keep a copy of the `colorNames` after the first render and then rely on the fact the `colorNames` should not change with `colorpalette` re-renders).

Conclusion: I think at the time we made this `colorPalette` re-render, it was really the easy/lazy way because there was no way to say to the widget "here is your new selected value". Since then, the widget has evolved a lot, maybe there are some stable-ok changes to do to replace the `_renderColorPalette` call in `setValue` by a `setSelectedColor` call, which would be efficient and synchronous code. Could you investigate? And even if not efficient enough for this PR goal, it would still be synchronous so your solution "only update it when opening the colorpicker" could still apply, but would be robust as no async part to handle in synchronous methods.

Reply:
Introduced a minimal placeholder widget, so that it can be registered as a child node upon construction then be replaced by an actual color palette on open - using the same mechanism as before this PR.

Reply:
What about not re-rendering like I suggested? Not possible?

Reply:
Introduced the suggested `setSelectedColor` call on color palette and on color picker. Works great !